app/assets/javascripts/discourse/components/d-editor.js.es6

```diff
@@ -301,7 +301,16 @@ export default Ember.Component.extend({
           showSelector({
             appendTo: self.$(),
             container,
-            onSelect: title => self._addText(self._getSelected(), `${title}:`)
+            onSelect: title => {
+              // Remove the previously type characters when a new emoji is selected from the selector.
+              let selected = self._getSelected();
+              let newPre = selected.pre.replace(/:[^:]+$/, ":");
+              let numOfRemovedChars = selected.pre.length - newPre.length;
+              selected.pre = newPre;
+              selected.start -= numOfRemovedChars;
+              selected.end -= numOfRemovedChars;
+              self._addText(selected, `${title}:`);
```

Comment:
Thanks for the PR @dtchau :) I don't think this is where the code broke though. Can you try a `git bisect` to and find out where it broke? Probably due to changes in autocomplete.js